p2p/src/peer_manager/peerdb/mod.rs

```diff
@@ -250,10 +343,92 @@ impl<S: PeerDbStorage> PeerDb<S> {
     }
 
     pub fn remove_outbound_address(&mut self, address: &SocketAddress) {
-        self.addresses.remove(address);
+        if !self.reserved_nodes.contains(address) {
+            self.addresses.remove(address);
+        }
+        self.address_tables.remove(address);
+
+        update_db(&self.storage, |tx| tx.del_known_address(address))
+            .expect("DB failure when removing known address {address}");
+    }
+
+    fn move_addr_to_tried(&mut self, address: &SocketAddress) {
+        let move_result = self.address_tables.move_to_tried(address);
+        let (addr_moved_to_new, discarded_addr) = match move_result {
+            None => (None, None),
+            Some((addr_moved_to_new, discarded_addr)) => (Some(addr_moved_to_new), discarded_addr),
+        };
+
+        debug_assert!(discarded_addr != Some(*address));
+
+        update_db(&self.storage, |tx| {
+            tx.add_known_address(address, KnownAddressState::Tried)?;
+
+            if let Some(addr_moved_to_new) = addr_moved_to_new {
+                tx.add_known_address(&addr_moved_to_new, KnownAddressState::New)?;
+            }
+
+            crate::Result::Ok(())
+        })
+        .expect("DB failure when updating known addresses}");
+
+        self.remove_from_addresses_if_some_non_reserved(discarded_addr);
```

Comment:
Might have missed something but a check that the tables contents is consistent with the persistent storage would be nice.

Reply:
The newly added tests call the `assert_addr_consistency` function which checks exactly this. Or do you mean I should add some checks to the production code too?

Reply:
> The newly added tests call the `assert_addr_consistency` function which checks exactly this. Or do you mean I should add some checks to the production code too?

That's fine, thanks